### PR TITLE
Add integrated search box, and other minor improvments

### PR DESCRIPTION
**Pull request by CatMe0w**
## Add integrated search box
image

## Move stop button to `ferrostar-core`
image

It looks more like iOS implementation now.

## Minor changes
- Implemented a proper way for serializing/deserializing Rust's `SystemTime` from JS.
- Removed animation when starting navigation to avoid initiating too many map tile requests.
- Other internal bug fixes.

### CODE REVIEW

File: web/src/ferrostar-core.ts [hunk 1]
```diff
@@ -113,6 +158,8 @@ export class FerrostarCore extends LitElement {
 
   // TODO: type
   async startNavigation(route: any, config: any) {
+    if (this.useIntegratedSearchBox) this.map?.removeControl(this.searchBox!);
```
Comment (ianthetechie) [inline web/src/ferrostar-core.ts:161]: Good idea making this configurable.

File: web/src/location.ts [hunk 2]
```diff
@@ -61,10 +61,7 @@ export class BrowserLocationProvider {
           courseOverGround: {
             degrees: Math.floor(position.coords.heading || 0),
           },
-          timestamp: {
-            secs_since_epoch: Math.floor(position.timestamp / 1000),
-            nanos_since_epoch: 0,
-          },
+          timestamp: Date.now(),
```
Comment (ianthetechie) [inline web/src/location.ts:64]: This seems like a regression 🤔

File: web/src/ferrostar-core.ts [hunk 3]
```diff
@@ -122,11 +169,7 @@ export class FerrostarCore extends LitElement {
           coordinates: route.geometry[0],
           horizontalAccuracy: 0.0,
           courseOverGround: null,
-          // TODO: find a better way to create the timestamp?
-          timestamp: {
-            secs_since_epoch: Math.floor(Date.now() / 1000),
-            nanos_since_epoch: 0,
-          },
+          timestamp: Date.now(),
```
Comment (ianthetechie) [inline web/src/ferrostar-core.ts:172]: Why do we always use the current timestamp?

File: web/src/ferrostar-core.ts [hunk 4]
```diff
@@ -88,6 +124,15 @@ export class FerrostarCore extends LitElement {
...
       zoom: 18,
     });
+
+    if (this.useIntegratedSearchBox) {
+      this.searchBox = new MapLibreSearchControl({
+        onResultSelected: (feature) => {
+          this.startNavigationFromSearch(feature.geometry.coordinates);
+        },
+      });
+      this.map.addControl(this.searchBox, "bottom-left");
```
Comment (ianthetechie) [inline web/src/ferrostar-core.ts:134]: I'd suggest this going in the top left instead (convention; that's where most such UIs put this).

**Review by ianthetechie — COMMENTED** [review]
One more general comment: I'm not sure what happened to the CSS for the map control. It normally looks great without any extra work.

Expected:

image

Current (also shows attribution even without results; that's unexpected):

image

**Comment by CatMe0w** [comment]
image
CSS fixed, and I moved all the text and buttons to the top since leaving the simulate navigation button on the bottom while the search box is on the top might be not intuitive. Is that ok?

**Pull request merged**